packages/shorebird_cli/lib/src/platform/apple.dart

```diff
@@ -289,12 +313,30 @@ $error''');
       );
     } on Exception catch (error) {
       linkProgress.fail('Failed to link AOT files: $error');
-      return (exitCode: ExitCode.software.code, linkPercentage: null);
+      return const LinkResult.failure();
     } finally {
       await dumpDebugInfo();
     }
+    Map<String, dynamic>? linkMetadata;
```

Comment:
```suggestion
    final Map<String, dynamic>? linkMetadata;
```

Reply:
We can't as I've currently written it, since the analyzer can't be sure it' isn't set twice.

Reply:
the `if (dumpDebugInfoDir == null)` branch seems to confuse it into believing it might be set twice.